app/services/edge_cache/bust_article.rb

```diff
@@ -1,87 +1,89 @@
 module EdgeCache
-  class BustArticle < Bust
+  class BustArticle < Buster
     def self.call(article)
       return unless article
 
       article.purge
 
-      bust(article.path)
-      bust("/#{article.user.username}")
-      bust("#{article.path}/")
-      bust("#{article.path}?i=i")
-      bust("#{article.path}/?i=i")
-      bust("#{article.path}/comments")
-      bust("#{article.path}?preview=#{article.password}")
-      bust("#{article.path}?preview=#{article.password}&i=i")
-      bust("/#{article.organization.slug}") if article.organization.present?
-      bust_home_pages(article)
-      bust_tag_pages(article)
-      bust("/api/articles/#{article.id}")
+      buster = EdgeCache::Buster.new
+
+      buster.bust(article.path)
+      buster.bust("/#{article.user.username}")
+      buster.bust("#{article.path}/")
+      buster.bust("#{article.path}?i=i")
+      buster.bust("#{article.path}/?i=i")
+      buster.bust("#{article.path}/comments")
+      buster.bust("#{article.path}?preview=#{article.password}")
+      buster.bust("#{article.path}?preview=#{article.password}&i=i")
+      buster.bust("/#{article.organization.slug}") if article.organization.present?
+      bust_home_pages(buster, article)
+      bust_tag_pages(buster, article)
+      buster.bust("/api/articles/#{article.id}")
 
       return unless article.collection_id
 
       article.collection.articles.find_each do |collection_article|
-        bust(collection_article.path)
+        buster.bust(collection_article.path)
       end
     end
 
-    def self.bust_home_pages(article)
+    def self.bust_home_pages(buster, article)
```

Comment:
I started to do this...and I realized it widens the scope of this PR quite a bit. We'd want to update all the cache busters and their specs would change with this setup. I think it would be "cleaner" in a separate PR since so much is changing in this one and caching touches the entire app.